backend/sirius-web-diagrams/src/main/java/org/eclipse/sirius/web/diagrams/components/NodeComponent.java

```diff
@@ -47,72 +51,104 @@ public NodeComponent(NodeComponentProps props) {
-            LabelComponentProps labelComponentProps = new LabelComponentProps(nodeVariableManager, labelDescription);
-            Element labelElement = new Element(LabelComponent.class, labelComponentProps);
-
-            INodeStyle style = nodeDescription.getStyleProvider().apply(nodeVariableManager);
-
-            // @formatter:off
-            var borderNodes = nodeDescription.getBorderNodeDescriptions().stream()
-                    .map(borderNodeDescription -> {
-                        var nodeComponentProps = new NodeComponentProps(nodeVariableManager, borderNodeDescription, true, cache);
-                        return new Element(NodeComponent.class, nodeComponentProps);
-                    })
-                    .collect(Collectors.toList());
-
-            var childNodes = nodeDescription.getChildNodeDescriptions().stream()
-                    .map(childNodeDescription -> {
-                        var nodeComponentProps = new NodeComponentProps(nodeVariableManager, childNodeDescription, false, cache);
-                        return new Element(NodeComponent.class, nodeComponentProps);
-                    })
-                    .collect(Collectors.toList());
-            // @formatter:on
-
-            List<Element> nodeChildren = new ArrayList<>();
-            nodeChildren.add(labelElement);
-            nodeChildren.addAll(borderNodes);
-            nodeChildren.addAll(childNodes);
-
+        if (nodeDescription.isSynchronised()) {
             // @formatter:off
-            NodeElementProps nodeElementProps = NodeElementProps.newNodeElementProps(nodeId)
-                    .type(type)
-                    .targetObjectId(targetObjectId)
-                    .targetObjectKind(targetObjectKind)
-                    .targetObjectLabel(targetObjectLabel)
-                    .descriptionId(nodeDescription.getId())
-                    .borderNode(isBorderNode)
-                    .style(style)
-                    .position(Position.UNDEFINED)
-                    .size(Size.UNDEFINED)
-                    .children(nodeChildren)
-                    .build();
+            nodeDescription.getSemanticElementsProvider().apply(variableManager)
+                .forEach(semanticElement -> children.add(this.renderNodeElementProps(semanticElement, Optional.empty())));
             // @formatter:on
-            Element nodeElement = new Element(NodeElementProps.TYPE, nodeElementProps);
-            children.add(nodeElement);
-
-            cache.put(nodeDescription.getId(), nodeElement);
-            cache.put(semanticElement, nodeElement);
+        } else {
+            Optional<Map> optionalTargetSemanticObjects = variableManager.get(DiagramDescription.TARGET_SEMANTIC_OBJECTS, Map.class);
+            if (optionalTargetSemanticObjects.isPresent()) {
+                Map<String, Object> targetSemanticObjects = optionalTargetSemanticObjects.get();
+               // @formatter:off
+               this.props.getPreviousNodes()
```

Comment:
I am not sure that I understand your algorithm. With your strategy, if two semantic elements had their order changed, you would still render them with the old order.